spec/requests/social_previews_spec.rb

```diff
@@ -6,31 +6,70 @@
   let(:organization) { create(:organization) }
   let(:article) { create(:article, user_id: user.id) }
 
+  before do
+    stub_request(:post, /hcti.io/).
+      to_return(status: 200,
+                body: '{ "url": "https://hcti.io/v1/image/6c52de9d-4d37-4008-80f8-67155589e1a1" }',
+                headers: { "Content-Type" => "application/json" })
+  end
+
   describe "GET /social_previews/article/:id" do
     it "renders proper article title" do
       get "/social_previews/article/#{article.id}"
       expect(response.body).to include CGI.escapeHTML(article.title)
     end
+
+    it "renders shecoded template when tagged with shecoded" do
+      she_coded_article = create(:article, tags: "shecoded")
+
+      get "/social_previews/article/#{she_coded_article.id}"
+
+      expect(response).to render_template(:shecoded)
+    end
+
+    it "requesting png generates an image and redirects to image url" do
```

Comment:
"requesting png ..." reads a bit funny in these descriptions of the tests, wonder if it's more clear if you just omit it